### PR TITLE
Landing page

### PR DESCRIPTION
**Pull request by thatsddr**
### What's New?

- New simple test files to check if the components render correctly;
- Added Landing Page with "Welcome", "Mission", and "About" Components;
- Added "UIContext" to manage the color mode and in the future also other UI-related variables;
- Added Footer and Header;
- Added customizable Meta component;
- Added TS Types for the new components;
- Custom Styles are now in one postCSS file (postCSS and tailwindCSS config files also added);
- Added Icons;
- README file matches the one in the main branch;

### Running the tests

1. Install the dependencies with ```npm install```
2. Run the tests with ```npm run tests```
3. Since the tests don't cover css and static files also start the development server with ```npm run dev``` and then visit http://localhost:3000

### CODE REVIEW

**Review by Darjusch — APPROVED** [review]
Try to make pull requests for features / tasks from the backlog to make it easier to review. 
Other wise it looks good!

File: components/landingPage/Mission.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import Image from "next/image";
+import InfoCard from "./InfoCard";
+
+export default function Mission() {
+	/**
+	 * The "Our Mission" Section Of The Landing Page
+	 */
+	return (
+		<div className="normal-bg flex flex-col min-h-screen h-full" id="mission">
+			<h2 data-testid="mission-title" className="section-header">
```
Comment (jwdotpark) [inline components/landingPage/Mission.tsx:10]: i don't know this was intentional but padding-bottom is give 1.25rem but padding-top was none and this looks a bit asymmetrical, i don't know the tailwind syntax but tried to give padding-top: 1.25rem on ff devtool for balancing it out just as other header and it looks symmetrical. just let u know!

Comment (thatsddr) [reply components/landingPage/Mission.tsx:10]: It was intentional and part of the section-header class. Anyway making it symmetrical is not an hard task, if you think that is an improvement I can modify the style.

**Comment by jwdotpark** [comment]
it's just a preference but snappy smooth-transition on scrollTo could be nice? :D everything works/looks great already though!

File: components/landingPage/Mission.tsx (hunk 1 above)
Comment (jwdotpark) [reply components/landingPage/Mission.tsx:10]: Ah i see, no don't bother ~_~ i don't have any problem with that. i just wanted to ask

**Comment by thatsddr** [comment]
> Try to make pull requests for features / tasks from the backlog to make it easier to review.
> Other wise it looks good!

Thanks! I will do that next time!

**Comment by thatsddr** [comment]
> it's just a preference but snappy smooth-transition on scrollTo could be nice? :D everything works/looks great already though!

It is a good idea, I did not think about it. Do you think we can wait and add this and maybe other animations as an "imporvement" in the future or should we implement this now?

**Comment by jwdotpark** [comment]
> 
> 
> > it's just a preference but snappy smooth-transition on scrollTo could be nice? :D everything works/looks great already though!
> 
> It is a good idea, I did not think about it. Do you think we can wait and add this and maybe other animations as an "imporvement" in the future or should we implement this now?

Oh it can definitely wait, not really an important feature i guess..

**Comment by thatsddr** [comment]
Alright! Then I'm merging this pull request. Thanks a lot for the review guys!

**Pull request merged**